managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java

```diff
@@ -128,12 +129,35 @@ public Result getRootCert(UUID customerUUID, UUID rootCA) {
     }
   }
 
-  public Result list(UUID customerUUID) {
+	public Result list(UUID customerUUID) {
```

Comment:
@daniel-yb Recently gave a great suggestion on one of my diffs to check the `inUse` of a certificate in the model itself. 
So, add a function as following in the CertificateInfo model class:
```public boolean getInUse() {
   // Do the check for the cert being in use
}
```
What this will do is that it will automatically populate the `isUse` field for all objects of this class.
So we won't need all this excess code in both list and delete. You can directly check the field.

Reply:
@Arnav15 agree,  if this gonna be used in multiple places then moving it to the model makes sense. If not, Is the helper method is sufficient in the controller itself? @daniel-yb

Reply:
Hey @mahendranbhat, look at this [commit](https://github.com/yugabyte/yugabyte-db/commit/9b879be1d169d90131dfb62814b138d9256cad48). This is essentially achieving a very similar goal to what this PR is for. So I'd definitely recommend going with an approach similar to what I did. Move the `inUse` check to the model, add a new function in the Universe model to find universes with that certificate, and return if it exists, and then accordingly mark it as `inUse` or not. Ideally, we'd want to add foreign keys for all such kinds of checks, but I guess for now this should be good enough.